src/DynamicModel.hpp

```diff
@@ -2,6 +2,7 @@
 #define _DYNAMIC_MODEL_H_
 
 #include "DataTypes.hpp"
+#include <stdexcept>
```

Comment:
Then the headers need to be added to the .cpp files in which the runtime_error is present, not "globally" in the headers (which impacts the library's users as well).